jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/impl/GenericSqlDialect.java

```diff
@@ -45,41 +46,41 @@ public SchemaOrCatalogSupport supportsJdbcSchemas() {
     @Override
     public IdentifierCaseHandling getUnquotedIdentifierHandling() {
         final SchemaAdapterNotes adapterNotes = getContext().getSchemaAdapterNotes();
-        if (adapterNotes.isSupportsMixedCaseIdentifiers()) {
+        if (adapterNotes.supportsMixedCaseIdentifiers()) {
             // Unquoted identifiers are treated case-sensitive and stored mixed case
             return IdentifierCaseHandling.INTERPRET_CASE_SENSITIVE;
         } else {
-            if (adapterNotes.isStoresLowerCaseIdentifiers()) {
+            if (adapterNotes.storesLowerCaseIdentifiers()) {
                 return IdentifierCaseHandling.INTERPRET_AS_LOWER;
-            } else if (adapterNotes.isStoresUpperCaseIdentifiers()) {
+            } else if (adapterNotes.storesUpperCaseIdentifiers()) {
                 return IdentifierCaseHandling.INTERPRET_AS_UPPER;
-            } else if (adapterNotes.isStoresMixedCaseIdentifiers()) {
+            } else if (adapterNotes.storesMixedCaseIdentifiers()) {
                 // This case is a bit strange - case insensitive, but still stores it mixed case
                 return IdentifierCaseHandling.INTERPRET_CASE_SENSITIVE;
             } else {
-                throw new RuntimeException(
-                      "Unexpected quote behavior. Adapternotes: " + SchemaAdapterNotes.serialize(adapterNotes));
+                throw new UnsupportedOperationException("Unexpected quote behavior. Adapter notes: " //
+                      + SchemaAdapterNotesJsonConverter.getInstance().convertToJson(adapterNotes));
```

Comment:
Since this is appearing in a log output, a more eye-friendly representation than JSON would be better.
Implement `toString()` to provide human-readable adapter notes.

Reply:
fixed